pmlc/dialect/tile/transforms/constant_types.h

```diff
@@ -0,0 +1,19 @@
+// Copyright 2019, Intel Corporation
+
+#pragma once
+
+#include "mlir/Pass/Pass.h"
+
+#include "pmlc/dialect/eltwise/ir/ops.h"
+#include "pmlc/dialect/tile/ir/ops.h"
+
+namespace pmlc::dialect::tile {
+
+struct ConstantTypesPass : public mlir::FunctionPass<ConstantTypesPass> {
+  void runOnFunction() final;
+
+  mlir::FloatType floatx_;
+  mlir::IntegerType intx_;
```

Comment:
We should avoid the use of standard types here and only use either `ScalarType` or even the `DataType` enum here instead.